lib/ocpp/v201/smart_charging.cpp

```diff
@@ -48,8 +57,16 @@ std::string profile_validation_result_to_string(ProfileValidationResultEnum e) {
         return "ChargingSchedulePeriodUnsupportedNumberPhases";
     case ProfileValidationResultEnum::ChargingSchedulePeriodExtraneousPhaseValues:
         return "ChargingSchedulePeriodExtraneousPhaseValues";
+    case ProfileValidationResultEnum::ChargingSchedulePeriodPhaseToUseACPhaseSwitchingUnsupported:
+        return "ChargingSchedulePeriodPhaseToUseACPhaseSwitchingUnsupported";
```

Comment:
These aren't intended to be sent in the response - rather, they're meant to provide detailed information for logging and for providing a detailed response at the message handling level.

Reply:
ok; can you clarify when/where these will actually be sent in the response? Ideally, we would ensure that the errors are standardized, since OCPP does not define status codes or strings.

Reply:
The use of these are only for internal tracing. The code that will leverage them is in progress. It will be in the handler for `SetChargingProfileRequest` the goal was to simply allow for logging so that when an issue arises the community could easily find where a profile was failing ie if the result of validation failed with `ChargingSchedulePeriodPhaseToUseACPhaseSwitchingUnsupported` you simply have to search where that is set.

As for a standardized set of these code we were adding them as we did our development with an internal use mentality. We can bring this up in a working group if that would help but it does not seem like something that should be formally created.

Reply:
> We can bring this up in a working group if that would help
Yes, I think it would be worthwhile to bring this up in a working group, if only to figure out what the plans are

> but it does not seem like something that should be formally created.
The MREC project, also funded by JOET, is completely focused on formally creating such standards. We should make sure to incorporate their recommendations.